### PR TITLE
[YUNIKORN-1649] Use Junit reporter in all suites

### PR DESCRIPTION
**Pull request by wusamzong**
### What is this PR for?
Using Junit in all test suites.
Create a conditional statement to exclude the case that the test directory(/tmp/e2e-test-reports) has been created by `make e2e_test`

### What type of PR is it?
* [x] - Sub-task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1649/

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by wilfred-s** [comment]
wait with kicking off the workflow until YUNIKORN-1666 is fixed

File: test/e2e/app/app_suite_test.go [hunk 1]
```diff
@@ -32,6 +34,16 @@ func init() {
 }
 
 func TestApp(t *testing.T) {
+	ginkgo.ReportAfterSuite("TestBasicScheduling", func(report ginkgo.Report) {
```
Comment (pbacsko) [inline test/e2e/app/app_suite_test.go:37]: TestApp != TestBasicScheduling

**Review by pbacsko — CHANGES_REQUESTED** [review]
Some copy-paste issues.

Also, pls check the naming of the xml files for consistency, other suites use names like "TEST-resourcefairness.xml", "TEST-bin_packing_junit.xml", but not every suite follows this. This could potentially be addressed in a follow-up JIRA.

**Review by manirajv06 — COMMENTED** [review]
@wusamzong Please address the comments raised by @pbacsko. Every test suite should have appropriate names (in 2 places).

Also, other scope of this jira is to explore "Does Junit reporter library itself provide any options to create the directory if it doesn't exists?". This helps to avoid creating directory in Makefile. Having hard coded directory name itself in Makefile is error prone. So we need to see if there is a way to avoid it.

**Review by HuangTing-Yao — COMMENTED** [review]
@wusamzong Please address the comments above,  the other parts is ok to me.

**Comment by wusamzong** [comment]
@pbacsko , @manirajv06 , @HuangTing-Yao  Thanks for your review! I will make the change.

**Comment by wusamzong** [comment]
The Ginkgo file only mentions creating a temporary directory.
Therefore, I created a function to create a Junit directory in common.util, so that the directory can be guaranteed to be created before each report is generated. Wondering if this is appropriate?

File: test/e2e/predicates/predicates_suite_test.go [hunk 2]
```diff
@@ -51,9 +51,11 @@ var _ = AfterSuite(func() {
 
 func TestPredicates(t *testing.T) {
 	ginkgo.ReportAfterSuite("TestPredicates", func(report ginkgo.Report) {
-		err := reporters.GenerateJUnitReportWithConfig(
+		err := common.CreateJUnitReportDir()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = reporters.GenerateJUnitReportWithConfig(
 			report,
-			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "Predicates_junit.xml"),
```
Comment (craigcondit) [inline test/e2e/predicates/predicates_suite_test.go:56]: Should be `TEST-predicates_junit.xml`.

File: test/e2e/queue_quota_mgmt/queue_quota_mgmt_suite_test.go [hunk 3]
```diff
@@ -52,9 +52,11 @@ var _ = AfterSuite(func() {
 
 func TestStateAwareAppScheduling(t *testing.T) {
 	ginkgo.ReportAfterSuite("TestQueueQuotaMgmt", func(report ginkgo.Report) {
-		err := reporters.GenerateJUnitReportWithConfig(
+		err := common.CreateJUnitReportDir()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = reporters.GenerateJUnitReportWithConfig(
 			report,
-			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "QueueQuotaMgmt_junit.xml"),
+			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-QueueQuotaMgmt_junit.xml"),
```
Comment (craigcondit) [inline test/e2e/queue_quota_mgmt/queue_quota_mgmt_suite_test.go:59]: Should be `TEST-queue_quota_mgmt_junit.xml`

File: test/e2e/resource_fairness/resource_fairness_suite_test.go [hunk 4]
```diff
@@ -52,7 +52,9 @@ var _ = AfterSuite(func() {
 
 func TestResourceFairness(t *testing.T) {
 	ginkgo.ReportAfterSuite("Resource Fairness", func(report ginkgo.Report) {
```
Comment (craigcondit) [inline test/e2e/resource_fairness/resource_fairness_suite_test.go:54]: Should be `TestResourceFairness`.

File: test/e2e/resource_fairness/resource_fairness_suite_test.go [hunk 5]
```diff
@@ -52,7 +52,9 @@ var _ = AfterSuite(func() {
...
-		err := reporters.GenerateJUnitReportWithConfig(
+		err := common.CreateJUnitReportDir()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = reporters.GenerateJUnitReportWithConfig(
 			report,
 			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-resourcefairness.xml"),
```
Comment (craigcondit) [inline test/e2e/resource_fairness/resource_fairness_suite_test.go:59]: Should be `TEST-resource_fairness_junit.xml`.

File: test/e2e/spark_jobs_scheduling/spark_jobs_scheduling_suite_test.go [hunk 6]
```diff
@@ -50,7 +50,9 @@ var _ = AfterSuite(func() {
 
 func TestSparkJobs(t *testing.T) {
 	ginkgo.ReportAfterSuite("TestSparkJobs", func(report ginkgo.Report) {
-		err := reporters.GenerateJUnitReportWithConfig(
+		err := common.CreateJUnitReportDir()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = reporters.GenerateJUnitReportWithConfig(
 			report,
 			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-SparkJobs_junit.xml"),
```
Comment (craigcondit) [inline test/e2e/spark_jobs_scheduling/spark_jobs_scheduling_suite_test.go:57]: Should be `TEST-spark_jobs_junit.xml`.

File: test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go [hunk 7]
```diff
@@ -54,9 +54,11 @@ var _ = AfterSuite(func() {
 
 func TestStateAwareAppScheduling(t *testing.T) {
 	ginkgo.ReportAfterSuite("TestStateAwareAppScheduling", func(report ginkgo.Report) {
-		err := reporters.GenerateJUnitReportWithConfig(
+		err := common.CreateJUnitReportDir()
+		Ω(err).NotTo(gomega.HaveOccurred())
+		err = reporters.GenerateJUnitReportWithConfig(
 			report,
-			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "StateAwareAppScheduling_junit.xml"),
+			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-StateAwareAppScheduling_junit.xml"),
```
Comment (craigcondit) [inline test/e2e/state_aware_app_scheduling/state_aware_app_scheduling_suite_test.go:61]: Should be `TEST-stateaware_app_scheduling_junit.xml`.

File: test/e2e/queue_quota_mgmt/queue_quota_mgmt_suite_test.go [hunk 8]
```diff
@@ -52,9 +52,11 @@ var _ = AfterSuite(func() {
 
 func TestStateAwareAppScheduling(t *testing.T) {
```
Comment (craigcondit) [inline test/e2e/queue_quota_mgmt/queue_quota_mgmt_suite_test.go:53]: Method should be named `TestQueueQuotaMgmt`.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Some naming issues, otherwise looks ok.

**Comment by wusamzong** [comment]
@craigcondit Thanks for your review! I solved the name issues you mentioned

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**